src/services/LocationService.ts

```diff
@@ -0,0 +1,44 @@
+// Types taken from the w3c Geolocation API: https://w3c.github.io/geolocation-api/#geolocation_interface
+
+export interface GeolocationCoordinates {
+  accuracy: number;
+  altitude: number | null;
+  altitudeAccuracy: number | null;
+  heading: number | null;
+  latitude: number;
+  longitude: number;
+  speed: number | null;
+}
+
+export interface GeolocationPosition {
+  coords: GeolocationCoordinates;
+  timestamp: number;
+}
+
+// Reference for code values: https://developer.mozilla.org/en-US/docs/Web/API/GeolocationPositionError/code
+export interface GeolocationPositionError {
+  code: number;
+  message: string;
+}
+
+class LocationService {
+  /**
+   * Retrieves the user's current location using navigator.geolocation.
+   * Times out and calls the error callback after 3s.
+   */
+  public getCurrentLocation(): Promise<GeolocationPosition> {
+    return new Promise((resolve, reject) => {
+      if (navigator.geolocation) {
+        const onSuccess = (position: GeolocationPosition) => resolve(position);
```

Comment:
Ah so this will prompt user for location access too right

Reply:
Yeah! It should show up as the info window in your browser that says something like "Do you allow this website to access your location?"

Reply:
Ahh got it kk thanks